### PR TITLE
User recommendation

### PR DESCRIPTION
**Pull request by gracelungu**
#### What does this PR do?
Adds recommendation features to the project

#### Description of Task to be completed?
~ Adds the recommendation model
~ Adds the controllers and logic 
~ Write tests for different routes
~ Document the APIs

#### How should this be manually tested?
Under the documentation, the recommendation section gives more details on how this PR can be tested.

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: src/__tests__/routes/recommendations.test.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
+import request from 'supertest';
```
Comment (oesukam) [inline src/__tests__/routes/recommendations.test.js:1]: @gracelungu  Can you use 2 spaces instead of 4 from your `VSCode`?

**Comment by gracelungu** [comment]
[Users should be able to recommend one another](https://trello.com/c/lFauvBxu/15-users-should-be-able-to-recommend-one-another)

File: src/controllers/RecommendationController.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    /**
+     * Recommend a user
+     *
+     * @static
+     * @param {*} req
+     * @param {*} res
+     * @returns {object} recommendation
+     * @memberof RecommendationController
+     */
+    static async recommend(req, res) {
```
Comment (oesukam) [inline src/controllers/RecommendationController.js]: @gracelungu Can you use the same naming as for other controllers to keep conformity? `createRecommendation`

File: src/controllers/RecommendationController.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        const { currentUser, receiver } = req;
+        const { description } = req.body;
+
+        if (currentUser._id.toString() === receiver.id) {
```
Comment (oesukam) [inline src/controllers/RecommendationController.js]: According to the documentation, they advise to use `currentUser._id.equals(receiver.id)

File: src/controllers/RecommendationController.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+     *
+     * @static
+     * @param {*} req
+     * @param {*} res
+     * @returns {void} recommendation
+     * @memberof RecommendationController
+     */
+    static async approve(req, res) {
+        const { currentUser } = req;
+        const { recommendation: id } = req.params;
```
Comment (oesukam) [inline src/controllers/RecommendationController.js]: `recommendationId` to keep conformity

**Review by oesukam — CHANGES_REQUESTED** [review]

**Pull request merged**